src/Compilers/CSharp/Portable/Symbols/AnonymousTypes/AnonymousTypeDescriptor.cs

```diff
@@ -79,7 +79,7 @@ internal bool Equals(AnonymousTypeDescriptor other, bool ignoreCustomModifiersAn
             ImmutableArray<AnonymousTypeField> otherFields = other.Fields;
             for (int i = 0; i < count; i++)
             {
-                if (!myFields[i].Type.Equals(otherFields[i].Type, ignoreCustomModifiersAndArraySizesAndLowerBounds, ignoreDynamic))
+                if (!myFields[i].Type.Equals(otherFields[i].Type, comparison & ~TypeCompareKind.IgnoreTupleNames))
```

Comment:
I think it should propagate type comparison. Caller decided whether tuple names in types are important or not and passed the flag. We should just use it recursively.